gdk/commands/component/project_utils.py

```diff
@@ -129,6 +131,8 @@ def get_project_config_values():
     vars["component_build_config"] = component_build_config
     vars["bucket"] = bucket
     vars["region"] = region
+    vars["encrypt"] = encrypt
+    vars["metadata"] = metadata
```

Comment:
What are the different possible formats of `metadata` in the `gdk-config.json`? IMO, these extra keys should go in publish along with the bucket configuration as these are args applicable to the bucket artifacts. Please provide us with the suggested config file format in the description of the PR.

Reply:
The format would key/value pairs.  

My thought process around this was to keep the bucket / artifact configuration together by putting them into the `gdk-config.json` to keep it close to my code base. Similar to how we do infrastructure as code.

Reply:
I think it would be beneficial to be able to do it both ways like how the gdk bucket is handled.